lib/ruby_lsp/test_unit_test_runner.rb

```diff
@@ -0,0 +1,98 @@
+# typed: true
+# frozen_string_literal: true
+
+require "test/unit"
+require "test/unit/ui/testrunner"
+require "ruby_lsp/test_reporter"
+require "ruby_indexer/lib/ruby_indexer/uri"
+
+module RubyLsp
+  class TestRunner < ::Test::Unit::UI::TestRunner
+    #: (::Test::Unit::TestCase test) -> void
+    def test_started(test)
+      current_test = test
+      @current_uri = uri_for_test(current_test)
+      return unless @current_uri
+
+      @current_test_id = "#{current_test.class.name}##{current_test.method_name}"
+      TestReporter.start_test(
+        id: @current_test_id,
+        uri: @current_uri,
+      )
+    end
+
+    #: (::Test::Unit::TestCase test) -> void
+    def test_finished(test)
+      if test.passed?
+        # tests with an Omission are still marked as passed, which seems strange
+        # return if test.instance_variable_get("@_result").faults.any?
+
+        TestReporter.record_pass(
+          id: @current_test_id,
+          uri: @current_uri,
+        )
+      end
+    end
+
+    #: (::Test::Unit::Failure | ::Test::Unit::Error | ::Test::Unit::Pending result) -> void
+    def result_fault(result)
+      case result
+      when ::Test::Unit::Failure
+        record_failure(result)
+      when ::Test::Unit::Error
+        record_error(result)
+      when ::Test::Unit::Pending
+        record_skip(result)
+      end
+    end
+
+    #: (::Test::Unit::Failure failure) -> void
```

Comment:
Can everything from this point down be private?

Reply:
It looks a bit odd but everything in the class can be `private`. (updated)